### ISSUE TITLE
How to rollback when scaling failed?

### ISSUE THREAD

**Issue by leavest** [apache/shardingsphere#16924]
## Question
How to rollback when scaling failed?

when scaling failed, run `preview select count(1) from t_order;`, return activeVersion for table rule.
And run `SHOW SHARDING TABLE RULE t_order;`, return failed newVersion for table rule.
When I stop the scaling and  start a new scaling, log print `There is an outstanding job with the same schema name`.

What's the current step to handle failed scaling which can return correct activeVersion for table when run `SHOW SHARDING TABLE RULE t_order;`?

Now, I have to delete all config and restart proxy for new test scaling.

**For English only**, other languages will not accept.

Before asking a question, make sure you have:

- Googled your question.
- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

**Comment by leavest** [comment]
我英文不是很好，我用中文写一下我的问题。
弹性伸缩失败时候怎么做从能让数据回到之前的状态？
目前的情况是，弹性伸缩失败时，执行`preview select count(1) from t_order;`，显示实际使用的规则。但是执行`SHOW SHARDING TABLE RULE t_order;`显示变更后的失败的没有生效的规则。

**Comment by sandynz** [comment]
Hi @leavest , thanks for your feedback. I tested on it, that's true, and it's confusing, we'll discuss about how to improve it.

About how to rollback, you could do following steps:
1. stop scaling job {jobId};
2. drop scaling job {jobId};
3. drop new version metadata in registry center manully.
4. clean created sharding databases and tables on target DB instance manually.

**Comment by sandynz** [comment]
Test SQLs:
```
mysql> preview select count(1) from t_order;
+------------------+-------------------------------------------------------------------------+
| data_source_name | actual_sql                                                              |
+------------------+-------------------------------------------------------------------------+
| ds_0             | select count(1) from t_order_0 UNION ALL select count(1) from t_order_2 |
| ds_1             | select count(1) from t_order_1 UNION ALL select count(1) from t_order_3 |
+------------------+-------------------------------------------------------------------------+
2 rows in set (0.31 sec)

mysql> SHOW SHARDING TABLE RULE t_order;
+---------+-------------------+---------------------+------------------------+--------------------------+----------------------------------+-----------------------------------+---------------------+-----------------------+-------------------------------+--------------------------------+---------------------+--------------------+---------------------+
| table   | actual_data_nodes | actual_data_sources | database_strategy_type | database_sharding_column | database_sharding_algorithm_type | database_sharding_algorithm_props | table_strategy_type | table_sharding_column | table_sharding_algorithm_type | table_sharding_algorithm_props | key_generate_column | key_generator_type | key_generator_props |
+---------+-------------------+---------------------+------------------------+--------------------------+----------------------------------+-----------------------------------+---------------------+-----------------------+-------------------------------+--------------------------------+---------------------+--------------------+---------------------+
| t_order |                   | ds_0,ds_1           |                        |                          |                                  |                                   | hash_mod            | order_id              | hash_mod                      | sharding-count=4               | order_id            | snowflake          | worker-id=123       |
+---------+-------------------+---------------------+------------------------+--------------------------+----------------------------------+-----------------------------------+---------------------+-----------------------+-------------------------------+--------------------------------+---------------------+--------------------+---------------------+
1 row in set (0.01 sec)

mysql> ALTER SHARDING TABLE RULE t_order(
    -> RESOURCES(ds_2,ds_3,ds_4),
    -> SHARDING_COLUMN=order_id,
    -> TYPE(NAME=hash_mod,PROPERTIES("sharding-count"=6)),
    -> KEY_GENERATE_STRATEGY(COLUMN=order_id,TYPE(NAME=snowflake))
    -> ), t_order_item(
    -> RESOURCES(ds_2,ds_3,ds_4),
    -> SHARDING_COLUMN=order_id,
    -> TYPE(NAME=hash_mod,PROPERTIES("sharding-count"=6)),
    -> KEY_GENERATE_STRATEGY(COLUMN=order_id,TYPE(NAME=snowflake))
    -> );
Query OK, 0 rows affected (0.61 sec)

mysql> SHOW SHARDING TABLE RULE t_order;
+---------+-------------------+---------------------+------------------------+--------------------------+----------------------------------+-----------------------------------+---------------------+-----------------------+-------------------------------+--------------------------------+---------------------+--------------------+---------------------+
| table   | actual_data_nodes | actual_data_sources | database_strategy_type | database_sharding_column | database_sharding_algorithm_type | database_sharding_algorithm_props | table_strategy_type | table_sharding_column | table_sharding_algorithm_type | table_sharding_algorithm_props | key_generate_column | key_generator_type | key_generator_props |
+---------+-------------------+---------------------+------------------------+--------------------------+----------------------------------+-----------------------------------+---------------------+-----------------------+-------------------------------+--------------------------------+---------------------+--------------------+---------------------+
| t_order |                   | ds_2,ds_3,ds_4      |                        |                          |                                  |                                   | hash_mod            | order_id              | hash_mod                      | sharding-count=6               | order_id            | snowflake          |                     |
+---------+-------------------+---------------------+------------------------+--------------------------+----------------------------------+-----------------------------------+---------------------+-----------------------+-------------------------------+--------------------------------+---------------------+--------------------+---------------------+
1 row in set (0.00 sec)

mysql> show scaling list;
+----------------------------------------------+----------------------+----------------------+--------+---------------------+-----------+
| id                                           | tables               | sharding_total_count | active | create_time         | stop_time |
+----------------------------------------------+----------------------+----------------------+--------+---------------------+-----------+
| 0130317c30317c3054317c7363616c696e675f646232 | t_order,t_order_item | 2                    | true   | 2022-04-24 09:48:12 | NULL      |
+----------------------------------------------+----------------------+----------------------+--------+---------------------+-----------+
1 row in set (0.05 sec)

mysql> preview select count(1) from t_order;
+------------------+-------------------------------------------------------------------------+
| data_source_name | actual_sql                                                              |
+------------------+-------------------------------------------------------------------------+
| ds_0             | select count(1) from t_order_0 UNION ALL select count(1) from t_order_2 |
| ds_1             | select count(1) from t_order_1 UNION ALL select count(1) from t_order_3 |
+------------------+-------------------------------------------------------------------------+
2 rows in set (0.01 sec)
```
`SHOW SHARDING TABLE RULE t_order;` works on latest version metadata.

**Comment by sandynz** [comment]
`SHOW SHARDING TABLE RULE t_order;` was fix by #17088
